main.tf

```diff
@@ -141,8 +141,13 @@ resource "aws_security_group" "this" {
   tags = "${var.tags}"
 }
 
+locals {
+  allowed_security_groups_count = "${length(var.allowed_security_groups)}"
+  allowed_cidr_blocks_count     = "${length(var.allowed_cidr_blocks)}"
+}
+
 resource "aws_security_group_rule" "default_ingress" {
-  count = "${var.allowed_security_groups_count}"
+  count = "${local.allowed_security_groups_count}"
```

Comment:
This won't work. In many situations it will give the famous `value of 'count' cannot be computed` error. That's why it's coded in this annoying way with the `xxx_count` variable, specifically to avoid this error.

Maybe TF 0.12 has addressed this but I'm not sure.

Reply:
I'll claim ignorance here: using a local, and declaring the local, above, allows you to actually count the elements of a list, which produces a number.

In which situation will you encounter `value of 'count' cannot be computed`?

Thanks!

Reply:
The local thing doesn't change the problem, just moves it. It happens sometimes when a resource attribute being counted and that resource isn't created yet.

e.g.

```hcl
allowed_security_groups         = ["sg-12345678", "${aws_security_group.xxxx.id}"]
```

Where `aws_security_group.xxxx.id` is not created yet. So first run will fail. You can work around it by running `terraform apply -target=aws_security_group.xxxx.id` and then `terraform apply` after.

Statement is here: https://github.com/hashicorp/terraform/issues/10857#issuecomment-268289775

And a longer issue is here: https://github.com/hashicorp/terraform/issues/12570

You can see the `_count` thing being used in other modules, e.g. https://github.com/terraform-aws-modules/terraform-aws-alb

Reply:
Ok, I've readded that variable and another for cidr blocks.

Reply:
@antonbabenko is this annoying `count` workaround resolved in 0.12 now?

Reply:
Yes, it looks like (at least to some degree if I understand it correctly, for example, хершы bug in ACM module](https://github.com/terraform-aws-modules/terraform-aws-acm/issues/10)).

I am updating modules to be as close as possible with existing features and after that, I am planning to update the code and do some breaking changes like this. I am looking forward to that myself :)

Reply:
@antonbabenko  looking forward too, because I can't access RDS from outside if I won't have additional cidr blocks in here to add my office IP :)

Reply:
> because I can't access RDS from outside if I won't have additional cidr blocks

You know you can just define an SG outside of the module with the required access and then pass it to the module as `allowed_security_groups`? That's the intended design.

Reply:
@max-rocket-internet we tried that, it defines EC2 sg, which is not RDS sg. Because RDS sg is ment to add a sg from EC2, which is basically ties VPC to RDS sg. And my intention is avoid VPC, and connect to RDS via DNS which AWS provides with "publicly_accessible" parameter :) In this case if you want this DNS to your, then source field in sg should be CIDR, and not another sg ID like it's done now. I'll just repeat in here - like if you provide your own SG to this module, it will create RDS SG and will link to your provided SG.  So this is not what I need at all :) I needs RDS SG to contain CIDR in source field.

Reply:
Thanks, Roman for the donation! Good surprise :) Will review and merge now.